lib/internal/child_process.js

```diff
@@ -230,12 +230,15 @@ util.inherits(ChildProcess, EventEmitter);
 
 
 function flushStdio(subprocess) {
-  if (subprocess.stdio == null) return;
-  subprocess.stdio.forEach(function(stream, fd, stdio) {
-    if (!stream || !stream.readable || stream._readableState.readableListening)
-      return;
+  if (subprocess.stdio === null || subprocess.stdio === undefined) return;
+
+  const stdio = subprocess.stdio;
```

Comment:
This could probably be moved to the top of the function, where the first conditional can take advantage of it.